lib/adapter.js

```diff
@@ -218,6 +224,8 @@ class MochaAdapter {
             this.runner.lastError = err
         }
 
+        message = this.generateUID(message)
```

Comment:
The structure of this method is not optimal. Can we rather have `generateUID` to return a parent and uid value and assign both to the message object here?